lib/msf/core/payload_generator.rb

```diff
@@ -159,6 +159,9 @@ def initialize(opts={})
       @encryption_iv = opts.fetch(:encryption_iv, nil)
 
       @framework  = opts.fetch(:framework)
+      if datastore.nil? || datastore.empty?
```

Comment:
`PayloadGenerator` already uses `blank?`, so you might as well.

Reply:
Sounds good, I'll update this now :)

Reply:
You might actually just want `nil?`. `{}` is a valid `datastore`, just empty, and it'll bail at option validation instead.

Reply:
Oh, it's using `fetch` with a default of `{}`.